druid-shell/src/platform/x11/util.rs

```diff
@@ -87,3 +91,38 @@ macro_rules! log_x11 {
         }
     };
 }
+
+/// A timer is a deadline (`std::Time::Instant`) and a `TimerToken`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub(crate) struct Timer {
+    deadline: Instant,
+    token: TimerToken,
+}
+
+impl Timer {
+    pub(crate) fn new(deadline: Instant) -> Self {
+        let token = TimerToken::next();
+        Self { deadline, token }
+    }
+
+    pub(crate) fn deadline(&self) -> Instant {
+        self.deadline
+    }
+
+    pub(crate) fn token(&self) -> TimerToken {
+        self.token
+    }
+}
+
+impl Ord for Timer {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // Ordering is so that earliest deadline sorts first
```

Comment:
Maybe worth pointing out here that we're putting it in a max-heap (because when I think "sorting", the first time that comes to mind is sorting vecs, and then the comment is confusing).